src/Acuminator/Acuminator.Tests/Tests/Utilities/AttributeInformation/AttributeInformationTests.cs

```diff
@@ -99,20 +109,64 @@ private void TestIsBoundAttribute(string source, List<bool> expected)
 
 				foreach (var attribute in attributes)
 				{
-					actual.Add(attributeInformation.IsBoundAttribute(attribute.AttributeClass));
+					actual.Add(attributeInformation.IsBoundAttribute(attribute));
 				}
 			}
 
 			Assert.Equal(expected, actual);
 		}
 
+		[Theory]
+		[EmbeddedFileData(@"PropertyIsDBBoundFieldAttribute.cs")]
+		public Task FieldBoundAttributesWithDynamicIsDBFieldSetInConstructorAsync(string source) =>
+			TestIsDBFieldPropertyAsync(source,
+								   new List<BoundType> { BoundType.DbBound, BoundType.Unbound });
+
+		[Theory]
+		[EmbeddedFileData(@"PropertyIsDBBoundFieldWithDefinedAttributes.cs")]
+		public Task FieldBoundAttributesWithDynamicIsDBFieldSetInAttributeDefinitionAsync(string source) =>
+		   TestIsDBFieldPropertyAsync(source,
+								  new List<BoundType> { BoundType.Unbound, BoundType.Unbound, BoundType.Unbound, BoundType.Unbound });
+
+		[Theory]
+		[EmbeddedFileData(@"PropertyIsDBBoundFieldWithoutDefinedAttributes.cs", internalCodeFileNames: new string[] { @"ExternalAttributes1.cs", @"ExternalAttributes2.cs" })]
+		public Task FieldBoundAttributesWithDynamicIsDBFieldSetInExternalAttributeDefinitionAsync(string source, string externalAttribute1, string externalAttribute2) =>
+		   TestIsDBFieldPropertyAsync(source,
+								  new List<BoundType> { BoundType.Unbound, BoundType.Unbound, BoundType.DbBound },
+								  new string[] { externalAttribute1, externalAttribute2 });
+
+		private async Task TestIsDBFieldPropertyAsync(string source, List<BoundType> expected, string[] code = null)
+		{
+			Document document = CreateDocument(source, externalCode: code);
+			SemanticModel semanticModel = await document.GetSemanticModelAsync().ConfigureAwait(false);
+			var syntaxRoot = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+			List<BoundType> actual = new List<BoundType>();
```

Comment:
@saloponov the capacity for actual could be set, It is expected that number of items in actual is equal to the number of items in expected. Therefore you can write:
`List<BoundType> actual = new List<BoundType>(capacity: expected.Count);`  

This way the internal array created for the List<T> will have size `expected.Count` from the beginning - less array copy and GC operations => better perfomance for the tests.